src/storage/src/wal.rs

```diff
@@ -78,10 +78,14 @@ impl<S: LogStore> Wal<S> {
         self.region_id
     }
 
-    #[cfg(test)]
     pub async fn close(&self) -> Result<()> {
-        let _ = self.store.stop().await;
-        Ok(())
+        self.store
```

Comment:
correctness: we should not stop logstore here. `Wal` only holds a reference to `Logstore`, it does not actually own it.   Otherwise closing a region will stop the logstore, which will make the whole storage engine malfunction.